### PR TITLE
clang-tidy & cppcheck - GUI

### PR DESCRIPTION
**Pull request by a-vinod**
We use bear to generate a compile database when the GUI build files are created. Cppcheck and clang-tidy then use this database to check these files. See the gui-tests CI job for the results of these two checks.

Pertains to #885

### CODE REVIEW

File: software/gui/gui.sh [hunk 1]
```diff
@@ -131,15 +134,15 @@ fi
 #########
 # BUILD #
 #########
-
 if [ "$1" == "--build" ]; then
+  cppcheck -ithird_party -ibuild .
```
Comment (martukas) [inline software/gui/gui.sh:138]: Should this be happening before everything else?
Especially before the root privileges check?
Should this also depend on the `--no-checks` option at all?

File: software/gui/gui.sh [hunk 2]
```diff
@@ -131,15 +134,15 @@ fi
...
 
   if [ "$EUID" -eq 0 ] && [ "$2" != "-f" ]; then
     echo "Please do not run build with root privileges!"
     exit 1
   fi
 
-  create_clean_directory build
+  7-create_clean_directory build
```
Comment (martukas) [inline software/gui/gui.sh:145]: That looks like a typo.

File: software/gui/gui.sh [hunk 3]
```diff
@@ -153,7 +156,23 @@ if [ "$1" == "--build" ]; then
...
+  pushd build && qmake $config_opt .. && bear make $j_opt && popd
+
+  cd build
+
+  if [ "$3" != "--no-checks" ]; then
+    set +e
+    set +o pipefail
+  fi
+
+  cppcheck --project=compile_commands.json -i../src/third_party .
```
Comment (martukas) [inline software/gui/gui.sh:168]: ok, I guess this means the one above is a duplicate?

**Review by martukas — CHANGES_REQUESTED** [review]
Little things.

File: software/gui/gui.sh (hunk 1 above)
Comment (a-vinod) [reply software/gui/gui.sh:138]: You're right, moving the `--no-checks` after the root check and then the cppcheck after that

File: software/gui/gui.sh (hunk 3 above)
Comment (a-vinod) [reply software/gui/gui.sh:168]: This second cppcheck uses the compile database to check the build files whereas the first one checks everything before building. I suppose then we need a clang-tidy check before building too for consistency.

Comment (a-vinod) [reply software/gui/gui.sh:168]: Nevermind. Looking into the CI job's output, it's checking those non-build files too.

**Review by martukas — APPROVED** [review]
I think this looks good now.

**Pull request merged**